### PR TITLE
Migrate batch score component

### PR DESCRIPTION
**Pull request by yetamsft**
This is the first PR to migrate batch-score components to azureml registry. In this PR, we copy all the code from internal batch-score ADO repo (up to the commit [a78eb0695a4d915fbdd27d7c8ee284501e1a1456](https://msdata.visualstudio.com/Vienna/_git/batch-score/commit/a78eb0695a4d915fbdd27d7c8ee284501e1a1456?refName=refs%2Fheads%2Fmain)) and fixed flake8 and pydocstyle errors.
We will setup PR gate for this component in the next PR.

Update:
Instead of inlining the environment (which is not allowed), we use an existing environment `azureml://registries/azureml/environments/model-evaluation/versions/15` to unblock before a dedicated environment is ready.

Here is a [test job](https://ml.azure.com/experiments/id/e7e78963-3e59-4957-b7e3-ce9f6e87e60d/runs/silver_pumpkin_n9gnc6cjq9?wsid=/subscriptions/c0afea91-faba-4d71-bcb6-b08134f69982/resourcegroups/batchscore-test-centralus/providers/Microsoft.MachineLearningServices/workspaces/ws-batchscore-centralus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#).

### CODE REVIEW

**Review by chupadhy — DISMISSED** [review]

**Review by vizhur — CHANGES_REQUESTED** [review]

File: assets/batch_score/components/driver/batch_score_components/v2/batch_score_llm/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    $[[--app_insights_connection_string ${{inputs.app_insights_connection_string}}]]
+    --ensure_ascii ${{inputs.ensure_ascii}}
+    --output_behavior ${{inputs.output_behavior}}
+    --initial_worker_count ${{inputs.initial_worker_count}}
+    --max_worker_count ${{inputs.max_worker_count}}
+    $[[--max_retry_time_interval  ${{inputs.max_retry_time_interval}}]]
+    --save_mini_batch_results ${{inputs.save_mini_batch_results}}
+    --mini_batch_results_out_directory ${{outputs.mini_batch_results_out_directory}}
+    --amlbi_async_mode ${{inputs.async_mode}}
+  environment:
```
[the remark this answers is not in the document]
Comment (yetamsft) [reply assets/batch_score/components/driver/batch_score_components/v2/batch_score_llm/spec.yaml:180]: Updated to use the latest openmpi cpu image. Do you think if I can use it image inline like this? Or is it actually required to register this environment?

[the remark this answers is not in the document]
Comment (vizhur) [reply assets/batch_score/components/driver/batch_score_components/v2/batch_score_llm/spec.yaml:180]: you need to register an environment, and that's a bit of a process

[the remark this answers is not in the document]
Comment (yetamsft) [reply assets/batch_score/components/driver/batch_score_components/v2/batch_score_llm/spec.yaml]: Yeah, I started 2 separate PRs for the image/environment.

[the remark this answers is not in the document]
Comment (yetamsft) [reply assets/batch_score/components/driver/batch_score_components/v2/batch_score_llm/spec.yaml]: Updated to use an existing environment instead.

**Review by chupadhy — APPROVED** [review]

**Review by vizhur — APPROVED** [review]

**Pull request merged**